### PR TITLE
Fix contact lists by removing filter limit

### PR DESCRIPTION
**Pull request by joshuatbrown**
## Issues covered
#1158

## Description
Removed the filter limit in `requestContactList` to ensure we get all contact lists so we can find the most recent one and use that.

This required refactoring so I could unit test RelayService, so this PR is a lot more than the 1 line that should fix the contact list bug.

## How to test
I have no idea. Help?

I've at least confirmed that there are no regressions, as best I can tell.

### CODE REVIEW

File: Nos/Service/Relay/RelayService.swift [hunk 1]
```diff
@@ -210,7 +210,6 @@ extension RelayService {
         let contactFilter = Filter(
             authorKeys: [authorKey],
             kinds: [.contactList],
-            limit: 1,
```
Comment (mplorentz) [inline Nos/Service/Relay/RelayService.swift:213]: Let's revert this change, see my other comment.

**Review by mplorentz — CHANGES_REQUESTED** [review]
I don't think this is the right approach. I think this will cause us to fetch and process a lot of extra contact lists, and they are already the most expensive type of nostr event to process. I also think the current code will leave these contact list subscriptions open indefinitely.

What we do on `main` (I think) is fetch up to 1 contact list (limit == 1) from any relay, then closing all subscriptions for the Filter. What we should do is fetch 1 contact list from _each_ relay **and then close the subscription**. What this code will do now is fetch all contact lists newer than the last one from each relay.

Uh oh. Now I am looking at the code and I think I have this wrong. 
> What we do on `main` (I think) is fetch up to 1 contact list (limit == 1) from any relay, then closing all subscriptions for the Filter.
I wrote this above but it looks incorrect to me. Looking at `RelayService.queueEventForParsing(...)` it looks like it only closes the given RelaySubscription when a matching event is received, not every RelaySubscription for the given Filter, which is what I thought it was doing.

How about this, can you revert the line I commented on that removes the `limit: 1` from the `contactFilter`? Then we can update the tests and merge those because those are super great. Then can you verify, either through the debugger or a test, that this is happening?
> What we should do is fetch 1 contact list from _each_ relay **and then close the subscription**.

This is a pretty weedy issue we essentially treat websockets two different ways: sometimes we want a observable stream of events that we leave open, and sometimes we just want to request a block of events and then close the connection like an HTTP request. In the code this is represented by the property `RelaySubscription.isOneTime`, but that seems poorly named now and kind of a hidden feature. If you think some simple renaming or refactoring can improve this lmk and maybe we can add it to this or another PR. My plan was to refactor it more heavily when we move to the gossip/outbox model. Also if you want to hop on a call to discuss this synchronously lmk.

File: Nos/Service/Relay/RelayService.swift (hunk 1 above)
Comment (joshuatbrown) [reply Nos/Service/Relay/RelayService.swift]: Done!

**Comment by joshuatbrown** [comment]
> How about this, can you revert the line I commented on that removes the `limit: 1` from the `contactFilter`? Then we can update the tests and merge those because those are super great.

This is done and ready for re-review. I'd be happy to merge these refactoring changes (plus tests) first then address the issue in a future PR.

**Review by mplorentz — APPROVED** [review]
